simSPI/linear_simulator/ctf_utils.py

```diff
@@ -0,0 +1,193 @@
+"""Module that multiplies ctf to fourier transform of the projection."""
+import numpy as np
+import torch
+import torch.fft
+
+
+class CTF(torch.nn.Module):
+    """Class that multiplies ctf to fourier transform of the projection.
+
+    Written by H. Gupta and J.N. Martel.
+
+    Parameters
+    ----------
+    config: class
+        Class containing parameters of the CTF
+
+    """
+
+    def __init__(self, config):
+        """Initialize image grid, wavelength, and frequency step per pixel."""
+        super(CTF, self).__init__()
+
+        self.config = config
+        self.wavelength = self._get_ewavelength()
+        self.frequency_step = 1.0 / (self.config.ctf_size * self.config.pixel_size)
+        n_half_len = float(self.config.ctf_size // 2)
+        ax = torch.arange(-n_half_len, n_half_len + self.config.ctf_size % 2)
+        mx, my = torch.meshgrid(ax, ax)
+        self.register_buffer("r2", mx ** 2 + my ** 2)
+        self.register_buffer("frequency", torch.sqrt(self.r2) * self.frequency_step)
+        self.register_buffer("angleFrequency", torch.atan2(my, mx))
+
+    def _get_ewavelength(self):
+        """Output the wavelength of the electron beam.
+
+        Returns
+        -------
+        wavelength: float
+            wavelength of the electron beam in A.
+        """
+        wavelength = 12.2639 / np.sqrt(
+            self.config.kv * 1e3 + 0.97845 * self.config.kv ** 2
```

Comment:
self.config.kv * (1e3 + 0.97845 * self.config.kv)

Reply:
Every little bit counts, right?

Reply:
Sorry, I think I missed replying to this comment in the previous PR. But I think the current version is more readable, at least for me.